packages/next/src/server/app-render/types.ts

```diff
@@ -57,13 +65,32 @@ export type FlightRouterState = [
   segment: Segment,
   parallelRoutes: { [parallelRouterKey: string]: FlightRouterState },
   url?: string | null,
-  /*
-  /* "refresh" and "refetch", despite being similarly named, have different semantics.
-   * - "refetch" is a server indicator which informs where rendering should start from.
-   * - "refresh" is a client router indicator that it should re-fetch the data from the server for the current segment.
-   *   It uses the "url" property above to determine where to fetch from.
+  /**
+   * "refresh" and "refetch", despite being similarly named, have different
+   * semantics:
+   * - "refetch" is used during a requets to inform the server where rendering
+   *   should start from.
+   *
+   * - "refresh" is used by the client to mark that a segmentshould re-fetch the
+   *   data from the server for the current segment. It uses the "url" property
+   *   above to determine where to fetch from.
+   *
+   * - "inside-shared-layout" is used during a prefetch request to inform the
+   *   server that even if the segment matches, it should be treated as if it's
+   *   within the "new" part of a navigation — inside the shared layout. If
+   *   the segment doesn't match, then it has no effect, since it would be
+   *   treated as new regardless. If it does match, though, the server does not
+   *   need to render it, because the client already has it.
+   *
+   *   A bit confusing, but that's because it has only one extremely narrow use
+   *   case — during a non-PPR prefetch, the server uses it to find the first
+   *   loading boundary beneath a shared layout.
+   *
+   *   TODO: We should rethink the protocol for dynamic requests. It might not
+   *   make for the client to send a FlightRouterState, since this type is
+   *   overloaded with concerns.
```

Comment:
👍

Reply:
```suggestion
   *   TODO: We should rethink the protocol for dynamic requests. It might not
   *   make sense for the client to send a FlightRouterState, since this type is
   *   overloaded with concerns.
```